cpp/src/arrow/python/pandas_convert.cc

```diff
@@ -176,6 +178,35 @@ Status AppendObjectStrings(StringBuilder& string_builder, PyObject** objects,
   return Status::OK();
 }
 
+Status AppendObjectFixedWidthBytes(FixedWidthBinaryBuilder& builder, PyObject** objects,
```

Comment:
google C++ recommends putting mutable args after input args, and making them pointers, e.g. `(..., FooBuilder* sink)`